src/components/Sections/GoogleMapsSection/index.tsx

```diff
@@ -0,0 +1,179 @@
+  @Prop({ required: false }) apikey!: GoogleMapsSectionProps["apikey"];
+  @Prop({ required: false })
+  startLocation!: GoogleMapsSectionProps["startLocation"];
+  @Prop({ required: false, default: 15 }) zoom!: GoogleMapsSectionProps["zoom"];
+  @Prop({ required: false, default: "en" })
+  language!: GoogleMapsSectionProps["language"];
+  @Prop({ required: false }) locations!: GoogleMapsSectionProps["locations"];
+  @Prop({ required: false, default: "Contact Us" })
+  buttonLabel!: GoogleMapsSectionProps["buttonLabel"];
+  @Prop({ required: true })
+  handleButtonClick!: GoogleMapsSectionProps["handleButtonClick"];
+
+  placeMarkers(map: google.maps.Map, ...locations: MapsLocation[]): void {
+    const renderDescriptionBox = (location: MapsLocation): Node => {
+      const template = `<h2 class="font-bold">${location.name}</h2>
+        <p>${location.street}</p>
+        ${location.description ? "<p>" + location.description + "</p>" : ""}
+        <p>${location.city}</p>
+        <button class="bg-black text-white hover:bg-gray-300 hover:text-black focus:outline-none p-2 my-4 w-auto overflow-hidden cursor-pointer font-light text-sm">
+        ${this.buttonLabel}
+        </button>`;
+      const div = document.createElement("div");
+      div.classList.add("w-32", "text-sm");
+      div.innerHTML = template;
+      div.querySelector("button")?.addEventListener("click", event => {
+        this.handleButtonClick(event, location);
+      });
+      return div;
+    };
+
+    locations.map((location: MapsLocation) => {
+      const marker = new google.maps.Marker({
+        position: location.position,
+        icon: markerIcon,
+        map,
+      });
+      const infoWindow = new google.maps.InfoWindow({
+        content: renderDescriptionBox(location),
+      });
+
+      marker.addListener("click", () => {
+        infoWindow?.open(map, marker);
+      });
+      return marker;
+    });
+  }
+
+  async initMap(): Promise<google.maps.Map> {
+    const center = this.startLocation || ({} as MapsPosition);
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(returnedLocation => {
+        center.lat = returnedLocation.coords.latitude;
+        center.lng = returnedLocation.coords.longitude;
+      });
+    }
+    const loader = new Loader({
+      apiKey: this.apikey,
+      version: "weekly",
+      language: this.language,
+    });
+    let map: google.maps.Map;
+    try {
+      await loader.load();
+      map = new google.maps.Map(document.getElementById("map") as HTMLElement, {
+        center,
+        zoom: this.zoom,
+        styles: DEFAULT_STYLE,
+      });
+    } catch (err) {
+      throw new Error(err);
+    }
+    return map;
+  }
+  mounted() {
+    if (this.language?.length !== 2) {
+      throw new Error("Language string must contain exactly 2 characters.");
+    }
+
+    this.initMap().then(
+      map => this.locations && this.placeMarkers(map, ...this.locations),
```

Comment:
We should change the function call to the following, so we don't need to destruct the array:
```suggestion
      map => this.locations && this.placeMarkers(map, this.locations),
```
Please have a look at my other comment on this file